modules/database/src/ioc/dbStatic/dbStaticRun.c

```diff
@@ -118,11 +118,13 @@ long dbAllocRecord(DBENTRY *pdbentry,const char *precordName)
         switch(pflddes->field_type) {
         case DBF_STRING:
             if(pflddes->initial)  {
-                if(strlen(pflddes->initial) >= pflddes->size) {
+                size_t initial_len = strlen(pflddes->initial);
+                if(initial_len >= pflddes->size) {
```

Comment:
I think there was, and still is, an off by one logic error in this test.  `strlen()` does not include the `'\n'` while `pflddes->size` should.  So the case `strlen()==size` would overrun by one byte.  I think the test should be

```c
if(strlen(pflddes->initial)+1 >= pflddes->size) {
```

That said, I don't think an overflow would occur for calcRecord as defined in epics-base.  I suspect there is some corruption, perhaps due to an incomplete rebuild after a source update.

Reply:
Also, this test would detect a misconfiguration of the record type.  Rather than repeating this test and error print in `dbAllocRecord()` for every record instance, it would be better to do so once in (or after) `registerRecordTypes()`.

Reply:
> I think there was, and still is, an off by one logic error in this test. `strlen()` does not include the `'\0'` while `pflddes->size` should. So the case `strlen()==size` would overrun by one byte. I think the test should be
> 
> ```c
> if(strlen(pflddes->initial)+1 >= pflddes->size) {
> ```
> 
> That said, I don't think an overflow would occur for calcRecord as defined in epics-base. I suspect there is some corruption, perhaps due to an incomplete rebuild after a source update.

If ```pflddes->size``` is 80 and ```strlen(pflddes->initial)``` is also 80, the ```>=```condition will prevent strcpy from being executed. Therefore, I believe the +1 is not necessary.

Reply:
`strlen(pflddes->initial)` should be 1 though for the CALC field of a calc record. Can you run the command `dbDumpField pdbbase calc CALC` at your IOC's shell? That prints the values of all the members of `pflddesc` from the running IOC. Your output should be the same as in my comment [below](https://github.com/epics-base/epics-base/pull/502#issuecomment-2163430204).

Reply:
I have verified using printf that the value of strlen(pflddes->initial) is 1.
```bash
stevek@ub24:~> softIoc -d calc.db
Starting iocInit
############################################################################
## EPICS R7.0.8.1-DEV
## Rev. R7.0.8-39-g57c14b49214aff76bbda
## Rev. Date Git: 2024-06-12 08:40:30 +0000
############################################################################
iocRun: All initialization complete
epics>
epics>
epics> dbDumpField pdbbase calc CALC
recordtype(calc)
    CALC
                 prompt: Calculation
                  extra:
              indRecordType: 49
                special: 103 SPC_CALC
             field_type: DBF_STRING
        process_passive: 1
               property: 0
                   base: 0
            promptgroup: 30 - Action
               interest: 0
               as_level: 1
                initial: 0
                   size: 80
                 offset: 696
Attribute: name RTYP value calc
Attribute: name VERS value none specified
```